app/support/order_row_importer.rb

```diff
@@ -223,4 +224,8 @@ def validate_product_is_importable
   def validate_user
     add_error("Invalid username or email") if user.blank?
   end
+
+  def validate_notes
+    add_error("Invalid note, 100 characters max") if note.present? && note.size > 100
```

Comment:
I tried to avoid the duplication by using the order details created in `#add_product_to_order` (see below) but it seems ackward because ItemAdder will `create!` the order details instead of building them. I can't seem to work around this without some major refactoring.

Maybe we can DRY up the note validation by making a constant for the max size, but I the larger issue of missing validations will still exist. Thoughts?

```
def add_product_to_order
  begin
    @order_details = order.add(product, quantity, note: note)
  rescue ActiveRecord::RecordInvalid => e
    add_error(e.message)
    order.destroy
    return
  end

  purchase_order! unless order.purchased?
  backdate_order_details_to_complete!
end
```